src/components/LayoutTree.tsx

```diff
@@ -446,6 +452,55 @@ const TreeNode = ({
     return lastPatches?.some(patch => deleteActions.includes(patch.actions[0]))
   })
 
+  const [showLineEndFauxCaret, setShowLineEndFauxCaret] = useState(false)
+  const [showLineStartFauxCaret, setShowLineStartFauxCaret] = useState(false)
+
+  // Hide the faux caret when typing occurs.
+  editingValueStore.subscribe(() => {
+    if (isTouch && isSafari() && caretRef.current) {
+      caretRef.current.style.display = 'none'
+      setShowLineStartFauxCaret(false)
+      setShowLineEndFauxCaret(false)
+    }
+  })
+
+  // If the thought isCursor and edit mode is on, position the faux cursor at the point where the
+  // selection is created.
+  useEffect(() => {
+    if (isTouch && isSafari() && caretRef.current) {
+      if (editing && isCursor) {
+        // The selection ranges aren't updated until the end of the frame when the thought is focused.
+        setTimeout(() => {
+          if (caretRef.current) {
```

Comment:
Similar opportunity to short circuit. I would probably add `fadeThoughtRef` as well:

```ts
if (!caretRef.current || !fadeThoughtRef.current) return
```